flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -339,6 +339,12 @@ extends Actor with ActorLogMessages with ActorLogging {
           }
         }
 
+      /**
+       * Always deny locking of ResultPartitions.
+       */
+      case LockResultPartition(partitionID: IntermediateResultPartitionID) =>
```

Comment:
we need the producing execution attempt id as well to uniquely identify a result partition